src/common/util/userTimeout.spec.ts

```diff
@@ -0,0 +1,250 @@
+        })
+
+        it(
+            'should return the DEFAULT_SESSION_TIMEOUT if the XUI team accidentally does not set a DEFAULT Session Timeout via the' +
+                'configuration.',
+            () => {
+                const roles = ['pui-organisation-manager']
+
+                const roleGroupSessionTimeouts: RoleGroupSessionTimeout[] = []
+
+                expect(getUserSessionTimeout(roles, roleGroupSessionTimeouts)).toEqual(DEFAULT_SESSION_TIMEOUT)
+            },
+        )
+
+        /**
+         * The following should never happen but the production code should be resilient to this edge case.
+         */
+        it('should return the DEFAULT_SESSION_TIMEOUT if there are no User Roles.', () => {
+            const roles: string[] = []
+
+            const roleGroupSessionTimeouts = [
+                {
+                    idleModalDisplayTime: 10,
+                    pattern: 'organisation',
+                    totalIdleTime: 80,
+                },
+            ]
+
+            expect(getUserSessionTimeout(roles, roleGroupSessionTimeouts)).toEqual(DEFAULT_SESSION_TIMEOUT)
+        })
+
+        it(
+            'should give preference to Session Timeout patterns in a PRIORITY ORDER. A pattern nearer to the top of the list is' +
+                'higher priority.',
+            () => {
+                // Roles are sorted by sortUserRoles() which is a side effect within getUserSessionTimeout(), but we are
+                // showing the sorted roles here, as it's easier to understand what happens.
+                const roles = [
+                    'caseworker',
+                    'caseworker-divorce-financialremedy',
+                    'caseworker-divorce-solicitor',
+                    'caseworker-probate',
+                    'caseworker-probate-solicitor',
+                    'pui-user-manager',
+                    'pui-finance-manager',
+                ]
+
+                const roleGroupSessionTimeouts = [
+                    {
+                        idleModalDisplayTime: 10,
+                        pattern: 'pui-user-manager',
+                        totalIdleTime: 80,
+                    },
+                    {
+                        idleModalDisplayTime: 20,
+                        pattern: 'caseworker-probate',
+                        totalIdleTime: 200,
+                    },
+                ]
+
+                expect(getUserSessionTimeout(roles, roleGroupSessionTimeouts)).toEqual(roleGroupSessionTimeouts[0])
+            },
+        )
+    })
+
+    /**
+     * Should sort the User's Roles alphabetically. Why? So that a priority order can be given to the Session Timeout +
+     * configuration list.
+     *
+     * Example: If we want a PUI Session Timeout.toBe( given preference over another Session Timeout it would be further
+     * up the Session Timeout Configuration list.
+     */
+    describe('sortRolesAlphabetically()', () => {
```

Comment:
Again this is duplication of tests see [sortArray](https://github.com/hmcts/rpx-xui-node-lib/blob/master/src/common/util/sortArray.spec.ts#L3)